File: api/WebApi/Models/Database/Account/Transactions/Transaction.cs
using System;
using WebApi.Models.Database.Account;
using WebApi.Models.Database.Base;

namespace WebApi.Models.Database
{
    public class Transaction : BaseModel
    {
        public double Amount { get; set; }
        public string Description { get; set; }
        public Guid AccountId { get; set; }
        public DateTimeOffset TransactionDate { get; set; }
        public Guid AccountCategoryId { get; set; }
        public virtual AccountCategory AccountCategory { get; set; }
        public virtual Account.Account Account { get; set; }
        

        public Transaction(double amount, string description, DateTimeOffset transactionDate, Guid accountCategoryId, Guid accountId)
        {
            Amount = amount;
            Description = description;
            AccountId = accountId;
            TransactionDate = transactionDate;
            AccountCategoryId = accountCategoryId;
        }
    }
}

File: web/src/app/accounts/_models/transaction.ts
export class Transaction {
  id: string;
  amount: number;
  description: string;
  accountId: string;
  transactionDate: Date;
  accountCategoryId: string;
  accountCategory: AccountCategory;
  convertedAmount: number;
  transactionNote: TransactionNote;
}

export interface AccountCategory {
  id: string;
  name: string;
  type: number;
}


export interface TransactionNote {
  note: string;
}
